noise.js

```diff
@@ -95,6 +115,12 @@ module.exports = class NoiseState extends SymmetricState {
     this.rx = null
     this.tx = null
     this.hash = null
+
+    function hasPskToken (handshake) {
+      return handshake.reduce((xs, x) => {
+        return (Array.isArray(x) && x.indexOf(TOK_PSK) !== -1) || xs
+      }, false)
```

Comment:
.some? also move this func to the bottom of the file as its not using any state from the constructor

Reply:
Done.